packages/nvidia_nat_core/src/nat/front_ends/fastapi/routes/execution.py

```
@@ -0,0 +1,154 @@
+            response_url=(f"/executions/{record.execution_id}"
+                          f"/interactions/{record.pending_interaction.interaction_id}/response"),
+        )
+    if record.status == ExecutionStatus.OAUTH_REQUIRED and record.pending_oauth is not None:
+        return ExecutionAcceptedOAuth(
+            execution_id=record.execution_id,
+            status_url=status_url,
+            auth_url=record.pending_oauth.auth_url,
+            oauth_state=record.pending_oauth.oauth_state,
+        )
+
+    raise ValueError(f"Cannot build 202 response for execution status: {record.status}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)